src/main/java/org/sciborgs1155/robot/drive/gyro/Navx.java

```diff
@@ -0,0 +1,51 @@
+package org.sciborgs1155.robot.drive.gyro;
+
+import com.kauailabs.navx.frc.AHRS;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.wpilibj.SPI;
+
+public class Navx implements GyroIO {
+  private AHRS ahrs;
+
+  public Navx() {
+    ahrs = new AHRS(SPI.Port.kMXP);
+  }
+
+  public double getRate() {
+    return ahrs.getRate();
+  }
+
+  @Override
+  public Rotation2d getRotation2d() {
+    return ahrs
+        .getRotation2d(); // dunno if using the built in method is faster/better, but if not, this
+    // can be removed
+  }
+
+  @Override
+  public Rotation3d getRotation3d() {
+    return ahrs
+        .getRotation3d(); // dunno if using the built in method is faster/better, but if not, this
```

Comment:
This is good, their native getRotation3d method uses quaternions in its constructor, so it's faster